### DISCUSSION TITLE
Support for HA (high availability)

### DISCUSSION THREAD

**Discussion by 8osm** [pelican-dev/panel#10] [Ideas]
Please consider adding support for running multiple instances of the frontend. See https://grafana.com/docs/grafana/latest/alerting/set-up/configure-high-availability/ for a possible implementation

**Comment by The0mikkel** [comment]
Could especially be interesting to have it easily deployable on Kubernetes for more HA and advanced setups.

**Reply by 8osm** [reply]
Seems like Laravel has distributed scheduling support, I'll try to contribute this when I have the time
https://laravel.com/docs/11.x/scheduling#running-tasks-on-one-server

**Reply by The0mikkel** [reply]
Yes, Laravel in itself is quite set up for scalable use.  
You can setup session to use an external service (instead of file).  

Could be an interesting PR to work on!

**Comment by lancepioch** [comment]
We'll be getting a production ready docker image out for the Panel in the next couple months.

**Comment by KittyKatt** [comment]
I would be incredibly happy to have support for leveraging a Kubernetes cluster (or multiple clusters). That has been a pain in my side the whole time I used pterodactyl because the rest of my entire infra is ran on a k8s cluster.

**Reply by KittyKatt** [reply]
I literally found [kubectyl](https://github.com/kubectyl) a week ago and set it all up only to find it's been deprecated by the maintainer, so now I'm hoping Pelican can hop into that space.

**Reply by 8osm** [reply]
I've started  working on a k8s operator for ptero/pelican. The actual operator part isn't there yet but I managed to make wings and panel kind of work inside k8s.
https://github.com/8osm/birdpanel-operator/tree/main/hack

**Comment by parkervcp** [comment]
I already planned on supporting kubernetes and having something like a helm chart to run it.

**Comment by lancepioch** [comment]
Thanks, this should be something that can be supported via plugins!

**Comment by Claiyc** [comment]
any update on this? Especially the k8s part?

**Comment by Svenum** [comment]
I am interested in this as well

**Comment by ItzMrF** [comment]
Any news on this?  Would love to see a k8s deployment of the full infra rather than just the panel.
